### ISSUE TITLE
populate_macos_target sleep issue

### ISSUE THREAD

**Issue by SamimSdz** [myspaghetti/macos-guest-virtualbox#164]
Hi, first of all thanks for writing this script. It is very useful ❤️

I am on stage **populate_macos_target** (Catalina), second terminal completed his work but the first one as you see below stopped and doesn't response (it is on **sleep**).
![image](https://user-images.githubusercontent.com/29577616/75332269-58b94980-5899-11ea-8ae4-c6d0039269ae.png)

I also tried to close the terminal and try to boot on macos directly from VBox but every time I start the machine it shows disk utilities.

**Comment by SamimSdz** [comment]
@myspaghetti is there any workaround for booting up the OS ?

**Comment by myspaghetti** [comment]
Hi, I believe one of the earlier stages failed. Please type in the virtual machine's Terminal the command `ls -lh /Install*.app/Contents/SharedSupport` - the result should be the following:
```
total 15927688
-rwxr-xr-x  1 root  wheel   328B Feb 26 11:41 AppleDiagnostics.chunklist
-rwxr-xr-x  1 root  wheel   2.9M Feb 26 11:41 AppleDiagnostics.dmg
-rwxr-xr-x  1 root  wheel   2.0K Feb 26 11:41 BaseSystem.chunklist
-rwxr-xr-x  1 root  wheel   474M Feb 26 11:41 BaseSystem.dmg
-rw-r--r--  1 root  wheel   7.1G Feb 26 11:46 InstallESD.dmg
-rwxr-xr-x  1 root  wheel   1.4K Feb 26 11:46 InstallInfo.plist
-rwxr-xr-x  1 root  wheel   1.5K Feb 26 11:41 InstallInfo.plist.bak
-rwxr-xr-x  1 root  wheel   1.5K Feb 26 11:46 InstallInfo.plist.bak2
```

Additionally, the command `find /Install*.app -name startosinstall` should return `/Install macOS Catalina.app/Contents/Resources/startosinstall`

If these files aren't present, it means the `populate_virtual_disks` stage failed, and you should start the script from the beginning. If the files exist, I'll try to reproduce the issue.

**Comment by SamimSdz** [comment]
> Hi, I believe one of the earlier stages failed. Please type in the virtual machine's Terminal the command `ls -lh /Install*.app/Contents/SharedSupport` - the result should be the following:
> 
> ```
> total 15927688
> -rwxr-xr-x  1 root  wheel   328B Feb 26 11:41 AppleDiagnostics.chunklist
> -rwxr-xr-x  1 root  wheel   2.9M Feb 26 11:41 AppleDiagnostics.dmg
> -rwxr-xr-x  1 root  wheel   2.0K Feb 26 11:41 BaseSystem.chunklist
> -rwxr-xr-x  1 root  wheel   474M Feb 26 11:41 BaseSystem.dmg
> -rw-r--r--  1 root  wheel   7.1G Feb 26 11:46 InstallESD.dmg
> -rwxr-xr-x  1 root  wheel   1.4K Feb 26 11:46 InstallInfo.plist
> -rwxr-xr-x  1 root  wheel   1.5K Feb 26 11:41 InstallInfo.plist.bak
> -rwxr-xr-x  1 root  wheel   1.5K Feb 26 11:46 InstallInfo.plist.bak2
> ```
> 
> Additionally, the command `find /Install*.app -name startosinstall` should return `/Install macOS Catalina.app/Contents/Resources/startosinstall`
> 
> If these files aren't present, it means the `populate_virtual_disks` stage failed, and you should start the script from the beginning. If the files exist, I'll try to reproduce the issue.

Both command's results are correct and the same as what you mentioned

![image](https://user-images.githubusercontent.com/29577616/75344905-d89ede00-58b0-11ea-8413-d6073fab55ec.png)

**Comment by myspaghetti** [comment]
In that case you can simply execute `'/Install macOS Catalina.app/Contents/Resources/startosinstall' --volume "/Volumes/macOS"` and the installation will begin. Meanwhile I'll try to reproduce the issue.

**Comment by myspaghetti** [comment]
In that case you can simply execute `'/Install macOS Catalina.app/Contents/Resources/startosinstall' --volume "/Volumes/macOS"` and the installation will begin. Meanwhile I'll try to reproduce the issue. The other terminal copies the NVRAM files to the EFI System Partition, but they are not required to boot macOS.

**Comment by SamimSdz** [comment]
> In that case you can simply execute `'/Install macOS Catalina.app/Contents/Resources/startosinstall' --volume "/Volumes/macOS"` and the installation will begin. Meanwhile I'll try to reproduce the issue. The other terminal copies the NVRAM files to the EFI System Partition, but they are not required to boot macOS.

Nothing happens with that command !

![image](https://user-images.githubusercontent.com/29577616/75346284-9aef8480-58b3-11ea-9851-0554c3d292d5.png)

**Comment by myspaghetti** [comment]
What's the output of `ls /Volumes`?

**Comment by SamimSdz** [comment]
> What's the output of `ls /Volumes`?

![image](https://user-images.githubusercontent.com/29577616/75347045-2cabc180-58b5-11ea-9c3b-7bc624447942.png)

**Comment by myspaghetti** [comment]
Could you run `curl -F 'f:1=@/var/log/install.log' ix.io` and paste the output URL here? If the file is empty, post the output of `md5 '/Install macOS Catalina.app/Contents/Resources/startosinstall'`

**Comment by SamimSdz** [comment]
> Could you run `curl -F 'f:1=@/var/log/install.log' ix.io` and paste the output URL here? If the file is empty, post the output of `md5 '/Install macOS Catalina.app/Contents/Resources/startosinstall'`

Sure, here you are :

`curl -F 'f:1=@/var/log/install.log' ix.io` ==> http://ix.io/2cIH

`md5 '/Install macOS Catalina.app/Contents/Resources/startosinstall'` ==> 5af9fae58cafb648e069e4d19bf4a8a8

**Comment by myspaghetti** [comment]
Good news, the log and the `md5sum` are absolutely fine. Bad news, I have no idea why the installer doesn't start. If you want you can compare the `md5sum` of each file from the installer:
```
find /Install*.app -type f -exec md5 {} \; > /tmp/md5sums.txt
curl -F 'f:1=@/tmp/md5sums.txt' ix.io
```
You can diff the resulting file with [this list of known good installer files md5sums](https://github.com/myspaghetti/macos-guest-virtualbox/files/4257565/known_good_md5sums.txt). Run `diff known_good_md5sums.txt your_list_of_installer_md5sums.txt`
If there's no output, the files are identical and your installer is fine, which means the problem is with the base system, which I will not be able to troubleshoot.

If that's the case, start the script from the beginning if you haven't already. If the issue persists, try installing Mojave instead, and then upgrade to Catalina.

**Comment by SamimSdz** [comment]
> Good news, the log and the `md5sum` are absolutely fine. Bad news, I have no idea why the installer doesn't start. If you want you can compare the `md5sum` of each file from the installer:
> 
> ```
> find /Install*.app -type f -exec md5 {} \; > /tmp/md5sums.txt
> curl -F 'f:1=@/tmp/md5sums.txt' ix.io
> ```
> 
> You can diff the resulting file with [this list of known good installer files md5sums](https://github.com/myspaghetti/macos-guest-virtualbox/files/4257565/known_good_md5sums.txt). Run `diff known_good_md5sums.txt your_list_of_installer_md5sums.txt`
> If there's no output, the files are identical and your installer is fine, which means the problem is with the base system, which I will not be able to troubleshoot.
> 
> If that's the case, start the script from the beginning if you haven't already. If the issue persists, try installing Mojave instead, and then upgrade to Catalina.

output of diff command :
```
< MD5 (/Install macOS Catalina.app/Contents/SharedSupport/InstallESD.dmg) = 201097c767e41ac0f3d38091df44ac7b
---
> MD5 (/Install macOS Catalina.app/Contents/SharedSupport/InstallESD.dmg) = 7c1599c05569952b95a3f5c885f89387
```

**Comment by myspaghetti** [comment]
Seems like `InstallESD.dmg` on the virtual machine is corrupt. You can check if it's good on the host. If you're running macOS, it's `md5 Catalina_InstallESDDmg.pkg`, if you're running anything else it's probably `md5sum Catalina_InstallESDDmg.pkg`. If the `md5sum` is not `201097c767e41ac0f3d38091df44ac7b` then delete the file (but nothing else) and then run `./macos-guest-virtualbox.sh prepare_macos_installation_files create_install_vdi populate_virtual_disks create_macos_installation_files_viso populate_macos_target` or just run the script from the beginning, it's more or less the same.

If the `md5sum` is good,  then just delete the `Catalina_InstallESD.part*` files and run the above stages or the entire script.

**Comment by myspaghetti** [comment]
Seems like `InstallESD.dmg` on the virtual machine is corrupt. You can check if it's good on the host. If you're running macOS, it's `md5 Catalina_InstallESDDmg.pkg`, if you're running anything else it's probably `md5sum Catalina_InstallESDDmg.pkg`. If the `md5sum` is not `201097c767e41ac0f3d38091df44ac7b` then delete the file (but nothing else) and then run `./macos-guest-virtualbox.sh prepare_macos_installation_files create_install_vdi create_macos_installation_files_viso populate_virtual_disks populate_macos_target` or just run the script from the beginning, it's more or less the same.

If the `md5sum` is good,  then just delete the `Catalina_InstallESD.part*` files and run the above stages or the entire script.

**Comment by myspaghetti** [comment]
Seems like `InstallESD.dmg` on the virtual machine is corrupt. You can check if it's good on the host. If you're running macOS, it's `md5 Catalina_InstallESDDmg.pkg`, if you're running anything else it's probably `md5sum Catalina_InstallESDDmg.pkg`. If the `md5sum` is not `201097c767e41ac0f3d38091df44ac7b` then delete the `Catalina_InstallESDDmg.pkg` file and the `Catalina_InstallESD.part*` files (but nothing else) and then run `./macos-guest-virtualbox.sh prepare_macos_installation_files create_install_vdi create_macos_installation_files_viso populate_virtual_disks populate_macos_target` or just run the script from the beginning, it's more or less the same.

If the `md5sum` is good,  then just delete the `Catalina_InstallESD.part*` files and run the above stages or the entire script.

**Comment by SamimSdz** [comment]
@myspaghetti `Catalina_InstallESDDmg.pkg` file was damaged. I've ran installation completely from beginning and now everything works just fine.
Thanks for you amenability and quick responses, hope someday I can compensate your favor.

**Comment by proyan** [comment]
@myspaghetti 
First of all, Thanks a lot for creating and maintaining this wonderful work. You've helped a lot of people who prefer linux but like mac features.

I have the same issue that @SamimSdz had: The script reaches the final part of, but then sleeps. I followed the discussion untill `md5sum Catalina_InstallESDDmg.pkg`. When I run this, I get
```
proyan@tengusan:macos-virtualbox$ md5sum Catalina_InstallESDDmg.pkg
36ce5f90e87e8bf7281079140484bba9  Catalina_InstallESDDmg.pkg
```

So it is clear that the `Catalina_InstallESDDmg.pkg` does not match the md5sum that you have in mind. I followed your advice, and deleted this file and the Catalina_InstallESD.part* files and re-ran the full script. After the download was finished, I reran the check, and it had downloaded the same file again. Then I found from one of your discussions https://github.com/myspaghetti/macos-virtualbox/issues/227#issuecomment-634631664 that a new catalina version had came out in May, so the script always downloads the new version.

So I'm currently stuck at the last stage of the installation. I would really appreciate your feedback and help on this issue. Please let me know if we can simply revive this old git issue, or if I should create a new one.

Thanks,
Rohan

**Comment by ManuelFte** [comment]
I was running into this same issue and tried several solutions:

1. I ran the script several times, since the OP of [this issue](https://github.com/myspaghetti/macos-virtualbox/issues/335) said that it simply worked at the 3rd try. I tried 4 times.
2. I downloaded the old version of the script linked [here](https://github.com/myspaghetti/macos-virtualbox/issues/335#issuecomment-727233626) and tried with that one.
3. I downloaded `InstallESDDmg.pkg` manually and checked its MD5 sum. It was identical to the one the script had downloaded.

None of the above worked and I continued getting stuck in the final step. Then I simply deleted everything, cleared the system's cache using BleachBit, restarted the computer, and tried again from the beginning. This time it worked.

For reference, I'm on Manjaro KDE.
